modules/gdscript/editor/gdscript_docgen.cpp

```diff
@@ -0,0 +1,271 @@
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "gdscript_docgen.h"
+#include "../gdscript.h"
+
+using GDP = GDScriptParser;
+using GDType = GDP::DataType;
+
+static String _get_script_path(const String &p_path) {
+	return vformat(R"("%s")", p_path.get_slice("://", 1));
+}
+
+static String _get_class_name(const GDP::ClassNode &p_class) {
+	const GDP::ClassNode *curr_class = &p_class;
+	if (!curr_class->identifier) { // All inner classes have a identifier, so this is the outer class
+		return _get_script_path(curr_class->fqcn);
+	}
+
+	String full_name = curr_class->identifier->name;
+	while (curr_class->outer) {
+		curr_class = curr_class->outer;
+		if (!curr_class->identifier) { // All inner classes have a identifier, so this is the outer class
+			return vformat("%s.%s", _get_script_path(curr_class->fqcn), full_name);
+		}
+		full_name = vformat("%s.%s", curr_class->identifier->name, full_name);
+	}
+	return full_name;
+}
+
+static PropertyInfo _property_info_from_datatype(const GDType &p_type) {
+	PropertyInfo pi;
+	pi.type = p_type.builtin_type;
+	if (p_type.kind == GDType::CLASS) {
+		pi.class_name = _get_class_name(*p_type.class_type);
+	} else if (p_type.kind == GDType::ENUM && p_type.enum_type != StringName()) {
+		pi.type = Variant::INT; // Only int types are recognized as enums by the EditorHelp
+		pi.usage |= PROPERTY_USAGE_CLASS_IS_ENUM;
+		// Replace :: from enum's use of fully qualified class names with regular .
+		pi.class_name = String(p_type.native_type).replace("::", ".");
+	} else if (p_type.kind == GDType::NATIVE) {
+		pi.class_name = p_type.native_type;
+	}
+	return pi;
+}
+
+void GDScriptDocGen::generate_docs(GDScript *p_script, const GDP::ClassNode *p_class, bool p_inner) {
+	p_script->_clear_doc();
+
+	DocData::ClassDoc &doc = p_script->doc;
+
+	doc.script_path = _get_script_path(p_script->get_script_path());
+	if (!p_script->name.is_empty()) {
```

Comment:
Here, for the sake of readability, I would test only if it's empty, then apply `doc.name = doc.script_path`, else `doc.name = p_script->name`